utils/validator.go

```
@@ -0,0 +1,68 @@
+	"github.com/gin-gonic/gin/binding"
+	"github.com/go-playground/validator/v10"
+)
+
+var validate *validator.Validate
+
+// Register the custom validators for the gin binding
+func init() {
+	validate = validator.New()
+	validate.SetTagName("binding")
+	if v, ok := binding.Validator.Engine().(*validator.Validate); ok {
+		v.RegisterValidation("matriculation", MatriculationNumberValidator)
```

golangci-lint (1.64.8)

Error return value of `v.RegisterValidation` is not checked

(errcheck)